api/src/main/java/io/strimzi/api/kafka/model/kafka/tieredstorage/TieredStorage.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+package io.strimzi.api.kafka.model.kafka.tieredstorage;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import io.strimzi.api.kafka.model.common.UnknownPropertyPreserving;
+import io.strimzi.crdgenerator.annotations.Description;
+import lombok.EqualsAndHashCode;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Abstract baseclass for different representations of tieredStorage, discriminated by {@link #getType() type}.
+ */
+@JsonTypeInfo(
+    use = JsonTypeInfo.Id.NAME,
+    include = JsonTypeInfo.As.EXISTING_PROPERTY,
+    property = "type"
+)
+@JsonSubTypes({
+    @JsonSubTypes.Type(value = TieredStorageCustom.class, name = TieredStorage.TYPE_CUSTOM)}
+)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@EqualsAndHashCode
+public abstract class TieredStorage implements UnknownPropertyPreserving, Serializable {
+
+    private static final long serialVersionUID = 1L;
+    public static final String TYPE_CUSTOM = "custom";
+
+    private final Map<String, Object> additionalProperties = new HashMap<>(0);
+
+    @Description("Storage type, must be either 'custom'.")
```

Comment:
not a native English speaker but reading "either" I would expect an alternative alongside "custom" while right now it's the only option. Maybe something like "Storage type. Only 'custom' is supported at the moment" or similar could work? @scholzj wdyt?

Reply:
You are right, I probably only copy paste the sentence without reading it :(